idaes/commands/env_info.py

```diff
@@ -0,0 +1,50 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""Get info about the environment IDAES is running in and the IDAES version
+"""
+
+__author__ = "John Eslick"
+
+import click
+from idaes.core.util.env_info import EnvironmentInfo
+from idaes.commands import cb
+
+
+@cb.command(
+    name="environment-info",
+    help="Print information about idaes, OS, dependencies...")
+@click.option(
+    "--solver",
+    multiple=True,
+    help="Add solvers to list of solvers to check"
+)
+@click.option(
+    "--file",
+    default=None,
+    help="Write output ot a file"
+)
+def environment_info(solver, file):
+    info = EnvironmentInfo(addtional_solvers=solver).display_dict()
+    if file is None:
+        for k, v in info.items():
+            click.echo("")
+            click.echo(f"{k}")
+            for l, q in v.items():
+                click.echo(f"    {l}: {q}")
+    else:
+        with open(file, "w") as f:
+            for k, v in info.items():
+                f.write("\n")
+                f.write(f"{k}\n")
+                for l, q in v.items():
+                    f.write(f"    {l}: {q}\n")
```

Comment:
General suggestion: I think that having the tool return JSON output would be good, and could possibly come in handy in the future for e.g. ease of integration with CI tools, etc.

Since the data is already easily JSONable, adding an optional `--json` flag should be straightforward, and I'd even suggest considering making it the default since it's not too unreadable (especially with options such as  `indent=4`) compared to the pretty-printed output.

Reply:
I switched the --file option to --json, which now writes the file in json format.